core/src/test/scala/org/apache/spark/CheckpointSuite.scala

```diff
@@ -512,6 +512,21 @@ class CheckpointSuite extends SparkFunSuite with RDDCheckpointTester with LocalS
     assert(rdd.isCheckpointedAndMaterialized === true)
     assert(rdd.partitions.size === 0)
   }
+
+  runTest("checkpoint all marked RDDs") { reliableCheckpoint: Boolean =>
```

Comment:
can you add another test to show that if we don't set this flag then the parent RDD is not checkpointed?